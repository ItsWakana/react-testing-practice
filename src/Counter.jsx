import React, { useState } from "react";

const Counter = () => {

    const [count, setCount] = useState(0);

    const incrementCount = () => {

        setCount((prev) => prev + 1);
    }

    return (
        <>
            <h1 className="counter">Count:{count}</h1>
            <button 
            className="increment-btn"
            onClick={incrementCount}>
            Increment</button>
        </>
    )
}

export default Counter;