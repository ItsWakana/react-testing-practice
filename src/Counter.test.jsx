import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Counter from "./Counter";

describe("Counter component", () => {
    it("Increments counter", async () => {
        const user = userEvent.setup();
        let count = 0;

        const { container } = render(<Counter />);

        expect(container).toMatchSnapshot();
        
        const counterHeading = screen.getByRole("heading");
        const button = screen.getByRole("button", { name: "Increment" });

        expect(counterHeading).toHaveTextContent(`Count:${count}`);

        await user.click(button);
        count++;
        expect(counterHeading).toHaveTextContent(`Count:${count}`);

    });
});