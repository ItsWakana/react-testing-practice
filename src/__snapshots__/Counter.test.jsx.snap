// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Counter component > Increments counter 1`] = `
<div>
  <h1
    class="counter"
  >
    Count:
    0
  </h1>
  <button
    class="increment-btn"
  >
    Increment
  </button>
</div>
`;
